collector/msft_fsrmquota.go

```diff
@@ -0,0 +1,232 @@
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		SoftLimit: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "softlimit"),
+			"If True, the quota is a soft limit. If False, the quota is a hard limit. The default value is False. Optional (SoftLimit)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		Template: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "template"),
+			"A valid quota template name. Up to 1KB in size. Optional (Template)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+		MatchesTemplate: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "matchestemplate"),
+			"If True, the property values of this quota match those values of the template from which it was derived. (MatchesTemplate)",
+			[]string{"quotaPath","quotaTemplate"},
+			nil,
+		),
+	}, nil
+}
+
+// Collect sends the metric values for each metric
+// to the provided prometheus Metric channel.
+func (c *MSFT_FSRMQuotaCollector) Collect(ctx *ScrapeContext, ch chan<- prometheus.Metric) error {
+	if desc, err := c.collect(ch); err != nil {
+		log.Error("failed collecting msft_fsrmquota metrics:", desc, err)
+		return err
+	}
+	return nil
+}
+
+// MSFT_FSRMQuota docs:
+// - <add link to documentation here>
+type MSFT_FSRMQuota struct {
+	Name string
+
+	Path      string
+	PeakUsage uint64
+	Size      uint64
+	Usage     uint64
+	Description           string
+	Template              string
+	//Threshold             string
+	Disabled              bool
+	MatchesTemplate       bool
+	SoftLimit             bool
+
+}
+
+func (c *MSFT_FSRMQuotaCollector) collect(ch chan<- prometheus.Metric) (*prometheus.Desc, error) {
+	var dst []MSFT_FSRMQuota
+	q := queryAll(&dst)
+	
+	var count int
+
+	if err := wmi.QueryNamespace(q, &dst, "root/microsoft/windows/fsrm"); err != nil {
+		return nil, err
+	}
+	
+		for _, quota := range dst {
+		
+		count++
+		quotaPath := quota.Path
+		quotaTemplate := quota.Template
+		quotaDescription := quota.Description
+		
+	ch <- prometheus.MustNewConstMetric(
+		c.PeakUsage,
+		prometheus.GaugeValue,
+		float64(quota.PeakUsage),
+		quotaPath,
+		quotaTemplate,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.Size,
+		prometheus.GaugeValue,
+		float64(quota.Size),
+		quotaPath,
+		quotaTemplate,
+	)
+
+	ch <- prometheus.MustNewConstMetric(
+		c.Usage,
+		prometheus.GaugeValue,
+		float64(quota.Usage),
+		quotaPath,
+		quotaTemplate,
+	)
+	ch <- prometheus.MustNewConstMetric(
+		c.Description,
+		prometheus.GaugeValue,
+		1.0,
+		quotaPath,quotaTemplate,quotaDescription,
+	)
+	ch <- prometheus.MustNewConstMetric(
+		c.Template,
+		prometheus.GaugeValue,
+		1.0,
+		quotaPath,
+		quotaTemplate,
+	)
+	
+	if quota.Disabled {
+				ch <- prometheus.MustNewConstMetric(c.Disabled,
+					prometheus.GaugeValue, 1.0, quotaPath,quotaTemplate)
+	} else {
+				ch <- prometheus.MustNewConstMetric(c.Disabled,
+					prometheus.GaugeValue, 0.0, quotaPath,quotaTemplate)
+	}
```

Comment:
I think it makes sense to add a helper in, say, collector.go to convert bool to float64, and then this becomes
```go
ch <- prometheus.MustNewConstMetric(c.Disabled,
					prometheus.GaugeValue, boolToFloat(quota.Disabled), quotaPath,quotaTemplate)
```
Sounds reasonable?